File: Programmers/Lv0/2의 영역.py
"""
출처: 프로그래머스 코딩 테스트 연습,
https://school.programmers.co.kr/learn/courses/30/lessons/181894
"""
# 내 풀이

def solution(arr):
    result = []
    sub = []
    for x in arr:
        if x == 2 or 2 in result:
            if x == 2:
                if not len(sub) == 0:
                    result = result + sub
                    result.append(x)
                    sub = []
                else:
                    result.append(x)
            else:
                sub.append(x)

        else:
            continue

    return result if len(result) > 0 else [-1]

# 다른 사람 풀이
def solution(arr):
    if 2 not in arr:
        return [-1]
    return arr[arr.index(2) : len(arr) - arr[::-1].index(2)]

File: Programmer_Java/Lv0/2의 영역.java
// package Programmers;

// public class 2의 영역 {
  
// }

// 출처:프로그래머스,
// https://school.programmers.co.kr/learn/courses/30/lessons/181894

// 풀이과정 
/*
 * // 2 들어가 있는 리스트를 만든 후 
// 2가 나오긴 전까지 들어갈 리스트를 만든 후 2가 나오면 최종적으로 합쳐주는 방식

import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        
        List<Integer> before = new ArrayList<>();
        List<Integer> after = new ArrayList<>();
        for(int i =0; i < arr.length; i++){
            if(arr[i]==2){
                after.addAll(before);
                after.add(arr[i]);
                before.clear();
            
            }
            else{
                if(after.size() >= 1){
                    before.add(arr[i]);
                }
            }
        }
        // System.out.println(after);
        
        int[] answer = new int[after.size()]; 
        
        for(int t =0 ; t<after.size(); t++){
            answer[t] = after.get(t);
            
        }
        if(after.size()==0){
            return new int[] { -1 };
        }
        
        else{
            return answer;
        }
        
    
    }

}
 */

/*
 * // 2 들어가 있는 리스트를 만든 후 
// 2가 나오긴 전까지 들어갈 리스트를 만든 후 2가 나오면 최종적으로 합쳐주는 방식

import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        List<Integer> before = new ArrayList<>();
        List<Integer> after = new ArrayList<>();
        for(int i:arr){
            
        }
        return answer;
        
    }

}
 */